master/buildbot/reporters/github.py

```diff
@@ -126,50 +126,53 @@ def send(self, build):
         context = yield props.render(self.context)
 
         sourcestamps = build['buildset'].get('sourcestamps')
-
-        if not sourcestamps or not sourcestamps[0]:
+        if not sourcestamps:
             return
 
-        project = sourcestamps[0]['project']
+        for sourcestamp in sourcestamps:
+            project = sourcestamp['project']
 
-        branch = props['branch']
-        m = re.search(r"refs/pull/([0-9]*)/merge", branch)
-        if m:
-            issue = m.group(1)
-        else:
+            # If branch is None, so is issue
             issue = None
+            if 'branch' in props:
+                m = re.search(r"refs/pull/([0-9]*)/merge", props['branch'])
+                if m:
+                    issue = m.group(1)
+
+            repoOwner = None
+            repoName = None
+            if "/" in project:
+                repoOwner, repoName = project.split('/')
+            else:
+                giturl = giturlparse(sourcestamp['repository'])
+                if giturl:
+                    repoOwner = giturl.owner
+                    repoName = giturl.repo
 
-        if "/" in project:
-            repoOwner, repoName = project.split('/')
-        else:
-            giturl = giturlparse(sourcestamps[0]['repository'])
-            repoOwner = giturl.owner
-            repoName = giturl.repo
-
-        if self.verbose:
-            log.msg("Updating github status: repoOwner={repoOwner}, repoName={repoName}".format(
-                repoOwner=repoOwner, repoName=repoName))
-
-        for sourcestamp in sourcestamps:
             sha = sourcestamp['revision']
+            # If the scheduler specifies multiple codebases, don't bother updating
+            # the ones for which there is no revision
+            if not sha:
+                log.msg(
+                    'Skipped status update for codebase {codebase}, '
+                    'context "{context}", issue {issue}.'.format(
+                        codebase=sourcestamp['codebase'], issue=issue, context=context))
```

Comment:
this line **could** be formatted as one parameter per line w/ `context=...` having a trailing `,` for easier future reviews